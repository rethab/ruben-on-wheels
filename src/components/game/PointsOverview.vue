<template>
  <v-container class="px-0">
    <v-row>
      <v-col :cols="cols" v-for="(player, index) in players" :key="index">
        <v-card>
          <v-card-title
            :class="{
              'text-medium-emphasis': !active(index),
              'text-subtitle-1': true,
            }"
            >{{ player.name }}</v-card-title
          >
          <v-card-text>
            <v-progress-linear
              height="12"
              :class="{
                'mb-2': true,
                'text-medium-emphasis': !active(index),
              }"
              :color="active(index) ? 'orange' : 'grey'"
              style="white-space: nowrap"
              :model-value="player.motivation"
              max="100"
              >{{ player.motivation }} Motivation</v-progress-linear
            >
            <v-progress-linear
              height="12"
              :class="{ 'text-medium-emphasis': !active(index) }"
              :color="active(index) ? 'orange' : 'grey'"
              style="white-space: nowrap"
              :model-value="player.points"
              max="500"
              >{{ player.points }} Points</v-progress-linear
            >
          </v-card-text>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { useGameStore } from "@/stores/game";

export default defineComponent({
  computed: {
    players() {
      const { players, currentPlayerIndex: idx } = useGameStore();
      return [...players.slice(idx, players.length), ...players.slice(0, idx)];
    },
    currentPlayerIndex() {
      return useGameStore().currentPlayerIndex;
    },
    cols() {
      const { xs } = this.$vuetify.display;
      return xs ? 6 : 4;
    },
  },
  methods: {
    active(index: number): boolean {
      return index === 0;
    },
  },
});
</script>
