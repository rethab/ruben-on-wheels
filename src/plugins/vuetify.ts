import "vuetify/styles";
import { createVuetify } from "vuetify";
import { aliases, mdi } from "vuetify/iconsets/mdi";
import * as components from "vuetify/components";
import * as directives from "vuetify/directives";
import "@mdi/font/css/materialdesignicons.css";

export default createVuetify({
  icons: { defaultSet: "mdi", aliases, sets: { mdi } },
  components,
  directives,
});

// Export for test.
export { components, directives };
