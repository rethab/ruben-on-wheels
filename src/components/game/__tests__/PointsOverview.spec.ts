import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import PointsOverview from "../PointsOverview.vue";

import vuetify from "@/plugins/vuetify";

function mountComponent() {
  return mount(PointsOverview as any, {
    props: {
      players: [
        { name: "Alice", currentCity: "Zurich", motivation: 10, points: 131 },
        { name: "Bob", currentCity: "Basel" },
        { name: "Carl", currentCity: "Cologne" },
      ],
      currentPlayerIndex: 1,
    },
    global: {
      plugins: [vuetify],
    },
  });
}

describe("PointsOverview", () => {
  it("shows current player first", () => {
    const wrapper = mountComponent();

    expect(wrapper.findAll(".v-card-title")[0].text()).toContain("Bob");
    expect(wrapper.findAll(".v-card-title")[1].text()).toContain("Carl");
    expect(wrapper.findAll(".v-card-title")[2].text()).toContain("Alice");
  });

  it("current player is highlighted", () => {
    const wrapper = mountComponent();

    expect(
      wrapper.findAll(".v-card-text")[0].find(".bg-orange").exists()
    ).toBeTruthy();
    expect(
      wrapper.findAll(".v-card-text")[1].find(".bg-orange").exists()
    ).toBeFalsy();
    expect(
      wrapper.findAll(".v-card-text")[2].find(".bg-orange").exists()
    ).toBeFalsy();
  });

  it("shows attributes of player", () => {
    const wrapper = mountComponent();
    expect(wrapper.findAll(".v-card-title")[2].text()).toContain("Alice");
    expect(wrapper.findAll(".v-card-subtitle")[2].text()).toContain(
      "City: Zurich"
    );
    expect(wrapper.findAll(".v-card-text")[2].text()).toContain(
      "10 Motivation"
    );
    expect(wrapper.findAll(".v-card-text")[2].text()).toContain("131 Points");
  });
});
