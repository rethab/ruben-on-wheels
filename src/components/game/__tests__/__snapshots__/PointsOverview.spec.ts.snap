// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PointsOverview > renders full component 1`] = `
"<div class=\\"v-container px-0\\">
  <div class=\\"v-row\\">
    <div class=\\"v-col v-col-6\\">
      <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\">
        <!---->
        <div class=\\"v-card__loader\\">
          <div class=\\"v-progress-linear v-theme--light\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
            <!---->
            <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__indeterminate\\">
                <div class=\\"v-progress-linear__indeterminate long\\"></div>
                <div class=\\"v-progress-linear__indeterminate short\\"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <!---->
        <div class=\\"v-card-title text-subtitle-1 pb-0\\">Bob</div>
        <div class=\\"v-card-subtitle\\">City: Basel</div>
        <div class=\\"v-card-text pt-1\\">
          <div class=\\"v-progress-linear v-theme--light mb-2\\" style=\\"top: 0px; height: 15px; --v-progress-linear-height: 15px; left: 50%; transform: translateX(-50%); white-space: nowrap; overflow: hidden;\\" role=\\"progressbar\\" aria-hidden=\\"false\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"0\\">
            <!---->
            <div class=\\"v-progress-linear__background bg-orange\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"slide-x-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__determinate bg-orange\\" style=\\"width: 0%;\\"></div>
            </transition-stub>
            <div class=\\"v-progress-linear__content\\"> Motivation</div>
          </div>
          <div class=\\"v-progress-linear v-theme--light\\" style=\\"top: 0px; height: 15px; --v-progress-linear-height: 15px; left: 50%; transform: translateX(-50%); white-space: nowrap;\\" role=\\"progressbar\\" aria-hidden=\\"false\\" aria-valuemin=\\"0\\" aria-valuemax=\\"500\\" aria-valuenow=\\"0\\">
            <!---->
            <div class=\\"v-progress-linear__background bg-orange\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"slide-x-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__determinate bg-orange\\" style=\\"width: 0%;\\"></div>
            </transition-stub>
            <div class=\\"v-progress-linear__content\\"> Points</div>
          </div>
        </div>
        <!---->
        <!----><span class=\\"v-card__underlay\\"></span>
      </div>
    </div>
    <div class=\\"v-col v-col-6\\">
      <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\">
        <!---->
        <div class=\\"v-card__loader\\">
          <div class=\\"v-progress-linear v-theme--light\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
            <!---->
            <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__indeterminate\\">
                <div class=\\"v-progress-linear__indeterminate long\\"></div>
                <div class=\\"v-progress-linear__indeterminate short\\"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <!---->
        <div class=\\"v-card-title text-medium-emphasis text-subtitle-1 pb-0\\">Carl</div>
        <div class=\\"v-card-subtitle\\">City: Cologne</div>
        <div class=\\"v-card-text pt-1\\">
          <div class=\\"v-progress-linear v-theme--light mb-2 text-medium-emphasis\\" style=\\"top: 0px; height: 15px; --v-progress-linear-height: 15px; left: 50%; transform: translateX(-50%); white-space: nowrap; overflow: hidden;\\" role=\\"progressbar\\" aria-hidden=\\"false\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"0\\">
            <!---->
            <div class=\\"v-progress-linear__background bg-grey\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"slide-x-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__determinate bg-grey\\" style=\\"width: 0%;\\"></div>
            </transition-stub>
            <div class=\\"v-progress-linear__content\\"> Motivation</div>
          </div>
          <div class=\\"v-progress-linear v-theme--light text-medium-emphasis\\" style=\\"top: 0px; height: 15px; --v-progress-linear-height: 15px; left: 50%; transform: translateX(-50%); white-space: nowrap;\\" role=\\"progressbar\\" aria-hidden=\\"false\\" aria-valuemin=\\"0\\" aria-valuemax=\\"500\\" aria-valuenow=\\"0\\">
            <!---->
            <div class=\\"v-progress-linear__background bg-grey\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"slide-x-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__determinate bg-grey\\" style=\\"width: 0%;\\"></div>
            </transition-stub>
            <div class=\\"v-progress-linear__content\\"> Points</div>
          </div>
        </div>
        <!---->
        <!----><span class=\\"v-card__underlay\\"></span>
      </div>
    </div>
    <div class=\\"v-col v-col-6\\">
      <div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\">
        <!---->
        <div class=\\"v-card__loader\\">
          <div class=\\"v-progress-linear v-theme--light\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
            <!---->
            <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__indeterminate\\">
                <div class=\\"v-progress-linear__indeterminate long\\"></div>
                <div class=\\"v-progress-linear__indeterminate short\\"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <!---->
        <div class=\\"v-card-title text-medium-emphasis text-subtitle-1 pb-0\\">Alice</div>
        <div class=\\"v-card-subtitle\\">City: Zurich</div>
        <div class=\\"v-card-text pt-1\\">
          <div class=\\"v-progress-linear v-theme--light mb-2 text-medium-emphasis\\" style=\\"top: 0px; height: 15px; --v-progress-linear-height: 15px; left: 50%; transform: translateX(-50%); white-space: nowrap; overflow: hidden;\\" role=\\"progressbar\\" aria-hidden=\\"false\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"10\\">
            <!---->
            <div class=\\"v-progress-linear__background bg-grey\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"slide-x-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__determinate bg-grey\\" style=\\"width: 10%;\\"></div>
            </transition-stub>
            <div class=\\"v-progress-linear__content\\">10 Motivation</div>
          </div>
          <div class=\\"v-progress-linear v-theme--light text-medium-emphasis\\" style=\\"top: 0px; height: 15px; --v-progress-linear-height: 15px; left: 50%; transform: translateX(-50%); white-space: nowrap;\\" role=\\"progressbar\\" aria-hidden=\\"false\\" aria-valuemin=\\"0\\" aria-valuemax=\\"500\\" aria-valuenow=\\"26.200000000000003\\">
            <!---->
            <div class=\\"v-progress-linear__background bg-grey\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"slide-x-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__determinate bg-grey\\" style=\\"width: 26.200000000000003%;\\"></div>
            </transition-stub>
            <div class=\\"v-progress-linear__content\\">131 Points</div>
          </div>
        </div>
        <!---->
        <!----><span class=\\"v-card__underlay\\"></span>
      </div>
    </div>
  </div>
</div>"
`;
